server/src/main/java/org/elasticsearch/action/admin/cluster/stats/CCSUsageTelemetry.java

```diff
@@ -184,9 +186,43 @@ public String toString() {
                 + '\''
                 + ", count="
                 + count
-                + ", latency(mean)="
-                + latency.getMean()
+                + ", latency="
+                + took.toString()
                 + '}';
         }
+
+        public long getSkippedCount() {
+            return skippedCount;
+        }
+
+        public CCSTelemetrySnapshot.PerClusterCCSTelemetry getSnapshot() {
+            return new CCSTelemetrySnapshot.PerClusterCCSTelemetry(count, skippedCount, took.getValue());
+        }
+
+    }
+
+    // TODO: I wonder if it wouldn't be more correct if this lived on the Snapshot side,
+    // but SearchUsage does it this way so following the pattern here.
+    public CCSTelemetrySnapshot getCCSTelemetrySnapshot() {
```

Comment:
Does this method need to be `synchronized` (or least a subsection of this method)? Since we use locks to do the updates we may need to have a lock to grab the snapshot. On the other hand since (currently) we also have LongAdders in many places, perhaps we can avoid a lock here? That would need a careful analysis. 

One risk of not using locking here is if someone adds to the Telemetry object later and doesn't ensure it is thread safe on the snapshot side (e.g., using a long instead of a LongAdder).

Reply:
Good question. I didn't fully consider the concurrency aspect yet, but certainly adding `synchronized` would make it safe. I am not 100% sure it's required yet, since if we have read while we're updating worst that could happen is that we get slightly inconsistent data (i.e. total counter incremented, but some other counter isn't yet) or such, which is not a huge deal. But I need to look into it more carefully to be sure.